src/rust/ide/src/model/suggestion_database/example.rs

```diff
@@ -0,0 +1,93 @@
+//! A module with structures related to examples found in Suggestion Database.
+use crate::prelude::*;
+
+use crate::double_representation::definition::DefinitionName;
+use crate::double_representation::module;
+use crate::double_representation::definition;
+
+use parser::Parser;
+
+
+
+// ==============
+// === Errors ===
+// ==============
+
+#[allow(missing_docs)]
+#[derive(Clone,Copy,Debug,Fail)]
+#[fail(display = "Invalid example code.")]
+pub struct InvalidExample;
+
+
+
+// ===============
+// === Example ===
+// ===============
+
+/// Example is a labeled piece of code user can put into their graph to see and learn how to use
+/// the language.
+///
+/// If a user picks an example, its `code` should became a body of a new method defined in current
+/// module. On the scene the node calling this method should appear.
+#[allow(missing_docs)]
+#[derive(Clone,Debug,Default,Eq,PartialEq)]
+pub struct Example {
+    pub name          : String,
+    pub code          : String,
+    pub documentation : String,
+}
+
+impl Example {
+    /// Return the example name converted in such way, that it will be a valid identifier.
+    ///
+    /// #### Example
+    /// ```
+    /// use enso_prelude::*;
+    ///
+    /// use ide::model::suggestion_database::Example;
+    ///
+    /// let name    = "With Spaces and Strange $#%^& Characters.".to_owned();
```

Comment:
If this is meant to test full Unicode support, it might be worth to also add some non-ascii characters.